src/lib/icu/Compare.hxx

```diff
@@ -25,26 +25,54 @@ class IcuCompare {
 #endif
 
 	AllocatedString needle;
+	bool fold_case;
+	bool strip_diacritics;
 
 public:
 	IcuCompare():needle(nullptr) {}
 
-	explicit IcuCompare(std::string_view needle) noexcept;
+	explicit IcuCompare(std::string_view needle, bool fold_case, bool strip_diacritics) noexcept;
 
 	IcuCompare(const IcuCompare &src) noexcept
 		:needle(src
 			? AllocatedString(src.needle)
-			: nullptr) {}
+			: nullptr),
+		 fold_case(src.fold_case),
+		 strip_diacritics(src.strip_diacritics) {}
 
 	IcuCompare &operator=(const IcuCompare &src) noexcept {
 		needle = src
 			? AllocatedString(src.needle)
 			: nullptr;
+		fold_case = src.fold_case;
+		strip_diacritics = src.strip_diacritics;
 		return *this;
 	}
 
-	IcuCompare(IcuCompare &&) = default;
-	IcuCompare &operator=(IcuCompare &&) = default;
+	IcuCompare(IcuCompare &&src) noexcept
+		:needle(src
+			? AllocatedString(std::move(src.needle))
+			: nullptr),
+		 fold_case(src.fold_case),
+		 strip_diacritics(src.strip_diacritics)
+	{
+		src.needle = nullptr;
+		src.fold_case = false;
+		src.strip_diacritics = false;
+	}
+
+	IcuCompare &operator=(IcuCompare &&src) noexcept {
+		needle = src
+			? AllocatedString(std::move(src.needle))
+			: nullptr;
+		fold_case = src.fold_case;
+		strip_diacritics = src.strip_diacritics;
+
+		src.needle = nullptr;
+		src.fold_case = false;
+		src.strip_diacritics = false;
+		return *this;
+	}
```

Comment:
What's the point of this change?

Reply:
That comment was wrong, sorry - I meant the copy ctor+operator, not copy+move ctor.